src/controller/CommissioningWindowParams.h

```diff
@@ -0,0 +1,185 @@
+    // If provided, must be at least kSpake2p_Min_PBKDF_Salt_Length bytes
+    // and at most kSpake2p_Max_PBKDF_Salt_Length bytes in length.
+    // Note that this an overloaded optional arg function to support existing APIs.
+    CommissioningWindowPasscodeParams & SetSalt(Optional<ByteSpan> salt)
+    {
+        mSalt = salt;
+        return *this;
+    }
+
+    bool GetReadVIDPIDAttributes() const { return mReadVIDPIDAttributes; }
+    // Should the API internally read VID and PID from the device while opening the
+    // commissioning window.  If this argument is `true`, the API will read VID and PID
+    // from the device and include them in the setup payload passed to the callback.
+    CommissioningWindowPasscodeParams & SetReadVIDPIDAttributes(bool readVIDPIDAttributes)
+    {
+        mReadVIDPIDAttributes = readVIDPIDAttributes;
+        return *this;
+    }
+
+    Callback::Callback<OnOpenCommissioningWindow> * GetCallback() const { return mCallback; }
+    // The function to be called on success or failure of opening the commissioning window.
+    // This will include the SetupPayload generated from provided parameters.
+    CommissioningWindowPasscodeParams & SetCallback(Callback::Callback<OnOpenCommissioningWindow> * callback)
+    {
+        mCallback = callback;
+        return *this;
+    }
+
+private:
+    Optional<uint32_t> mSetupPIN                              = NullOptional;
+    Optional<ByteSpan> mSalt                                  = NullOptional;
+    bool mReadVIDPIDAttributes                                = false;
+    Callback::Callback<OnOpenCommissioningWindow> * mCallback = nullptr;
+};
+
+class CommissioningWindowVerifierParams : public CommissioningWindowCommonParams<CommissioningWindowVerifierParams>
+{
+public:
+    CommissioningWindowVerifierParams() = default;
+
+    ByteSpan GetVerifier() const { return mVerifier; }
+    // The PAKE passcode verifier generated with enclosed iterations, salt and not-enclosed passcode.
+    CommissioningWindowVerifierParams & SetVerifier(ByteSpan verifier)
+    {
+        mVerifier = verifier;
+        return *this;
+    }
+
+    ByteSpan GetSalt() const { return mSalt; }
+    // The salt that was used to generate the verifier.
+    // It must be at least kSpake2p_Min_PBKDF_Salt_Length bytes.
+    // Note: This is REQUIRED when verifier is used
+    CommissioningWindowVerifierParams & SetSalt(ByteSpan salt)
+    {
+        mSalt = salt;
+        return *this;
+    }
+
+    Callback::Callback<OnOpenCommissioningWindowWithVerifier> * GetCallback() const { return mCallback; }
+    // The function to be called on success or failure of opening the
+    // commissioning window. This will NOT include the SetupPayload.
+    CommissioningWindowVerifierParams & SetCallback(Callback::Callback<OnOpenCommissioningWindowWithVerifier> * callback)
+    {
+        mCallback = callback;
+        return *this;
+    }
+
+private:
+    ByteSpan mVerifier;
+    ByteSpan mSalt;
```

Comment:
This needs to have sentinels so we properly fail out if these never get initialized.

Reply:
Changed these bytespan members type to optional and made things go kaboom in accessors with `Value()`: https://github.com/project-chip/connectedhomeip/pull/34209/commits/da28dd643e7419092f7a073ab2011fda251f2504